docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,34 +1,85 @@
+
+Keep in mind that setting up your cluster to use cloud controller manager will change your cluster behaviour in a few ways:
+
+* kubelets specifying `--cloud-provider=external` will add a [taint during initialization](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/well_known_labels.go#L51-L55). This marks the node as needing a second initialization from an external controller before it can be scheduled work. Note that in the event that cloud controller manager is not available, new nodes in the cluster will be left unscheduable. The taint is important since the scheduler may require cloud specific information about nodes such as it's region or type (high cpu, gpu, high memory, spot instance, etc).
+* cloud information about nodes in the cluster will no longer be retrieved using local metadata, but instead all API calls to retreive node information will go through cloud controller manager. This may mean you can restrict access to your cloud API on the kubelets for better security. For larger clusters you may want to consider if cloud controller manager will hit rate limits since it is now responsible for almost all API calls to your cloud from within the cluster.
+
+
+As of v1.8, cloud controller manager can implement:
+
+* node controller - responsible for updating nodes using cloud APIs.
+* service controller - responsible for loadbalancers on your cloud against services of type LoadBalancer.
+* route controller - responsible for setting up network routes on your cloud
+* persistent volume label controller - responsible for labeling persistent volumes on your cloud - ensure that the persistent volume label admission plugin is not enabled on your kube-apiserver.
+
+
+## Examples
+
+If you are using a cloud that is currently supported in Kubernetes core and would like to adopt cloud controller manager, see the [cloud controller manager in kubernetes core](https://github.com/kubernetes/kubernetes/tree/master/cmd/cloud-controller-manager).
+
+For cloud controller managers not in Kubernetes core, you can find the respective projects in repos maintained by cloud vendors or sig leads.
+
+* [DigitalOcean](https://github.com/digitalocean/digitalocean-cloud-controller-manager)
+* [Rancher](https://github.com/rancher/rancher-cloud-controller-manager)
+
+
+## Developing your own Cloud Controller Manager
+
+### Background
+
+Before going into how to build your own cloud controller manager, some background on how it works under the hood is helpful. The cloud controller manager is code from `kube-controller-manager` utilizing Go interfaces to allow implementations from any cloud to be plugged in. Most of the logic for the cloud controller manager will still live in Kubernetes core but the implementations for each cloud can develop at any pace, so long as the [cloud provider interface](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L29-L50) is satisifed.
```

Comment:
Most of the _scaffolding_ and generic controller implementation will be in core; in the cloud controller _library_.
But it will always "exec out" to the interface you give to it

_OR_ you can dismiss that and write your ccm from scratch, your choice